### PR TITLE
Fix #13278 - Wrong return type in Salesrule Module RuleInterface.php

### LINKED ISSUE #13278: Wrong return type in Salesrule Module RuleInterface.php
### Preconditions

1.  Magento CE 2.2.2

### Steps to reproduce

1. creating a salesrule object via RuleInterfaceFactory and using its setIsActive() method before another setter method with method chaining.

### Expected result

1. Return typehint of setIsActive() of Magento\SalesRule\Api\Data\RuleInterface is RuleInterface type ($this).

### Actual result

1. Return typehint of setIsActive() of Magento\SalesRule\Api\Data\RuleInterface is bool type.

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This is fix for #13278 

### Fixed Issues (if relevant)

1. magento/magento2#13278: Wrong return type in Salesrule Module RuleInterface.php

### Manual testing scenarios (*)

1. Create salesrule object via RuleInterfaceFactory and use its setIsActive() method before another setter method with method chaining.
2. Return typehint of setIsActive() of Magento\SalesRule\Api\Data\RuleInterface should be RuleInterface type ($this).

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
There is something strange with tests happened here, while it shouldn't. I'll investigate it deeper, for now this PR on me

File: app/code/Magento/SalesRule/Api/Data/RuleInterface.php [hunk 1]
```diff
@@ -59,14 +61,14 @@ public function setName($name);
     /**
      * Get display label
      *
-     * @return \Magento\SalesRule\Api\Data\RuleLabelInterface[]|null
```
Comment (Stepa4man) [inline app/code/Magento/SalesRule/Api/Data/RuleInterface.php]: @Bartlomiejsz Please return back fully qualified return types and param definition in doc block. For future, please read [this](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/service-contracts/service-to-web-service.html#service-interface-requirements)

**Review by Stepa4man — CHANGES_REQUESTED** [review]

File: app/code/Magento/SalesRule/Api/Data/RuleInterface.php (hunk 1 above)
Comment (Bartlomiejsz) [reply app/code/Magento/SalesRule/Api/Data/RuleInterface.php]: @Stepa4man clear. done now

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @Bartlomiejsz,
Could you review and fix failing static tests?

**Comment by Bartlomiejsz** [comment]
Hi @ihor-sviziev, done

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6044](https://jira.corp.magento.com/browse/ENGCOM-6044) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Review by Stepa4man — APPROVED** [review]

**Pull request merged**